<?xml version="1.0" encoding="UTF-8" ?>
<launch>

        <!--<rosparam name="use_sim_time" value="true" />-->

        <!-- overwriting these args -->
        <arg name="debug" default="false" />
        <arg name="gui" default="true" />
        <arg name="paused" default="false" />
        <arg name="use_sim_time" default="true" />
        <arg name="headless" default="false" />
        <arg name="model" default="model2.urdf"/>

        <!-- include gazebo_ros launcher -->        
        <include file="$(find gazebo_ros)/launch/empty_world.launch">
                <arg name="headless" value="$(arg headless)" />
                <arg name="debug" value="$(arg debug)" />
                <arg name="gui" value="$(arg gui)" />
                <arg name="paused" value="$(arg paused)" />
                <arg name="use_sim_time" value="true" />
        </include>
        
        
        <!-- set parameters -->
        <param name="robot_description" textfile="$(find quadrupedal_tests)/urdf/$(arg model)" />

        <!-- start urdf spawner -->
        <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" args="-x 0 -y 0 -z .8 -unpause -urdf -model quadrupedal_robot -param robot_description" respawn="false" output="screen" />

        <rosparam file="$(find quadrupedal_tests)/config/quadrupedal_control.yaml" command="load" />

        <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"  ns="/quadrupedal_test"
        args="left_front_leg_shoulder_rotate_controller right_front_leg_shoulder_rotate_controller left_back_leg_shoulder_rotate_controller right_back_leg_shoulder_rotate_controller left_front_leg_shoulder_hinge_controller right_front_leg_shoulder_hinge_controller left_back_leg_shoulder_hinge_controller right_back_leg_shoulder_hinge_controller left_front_leg_elbow_hinge_controller right_front_leg_elbow_hinge_controller left_back_leg_elbow_hinge_controller right_back_leg_elbow_hinge_controller joint_state_controller" />

        <!-- start joint_state_publisher, robot_state_publisher, rviz-->

        <!--<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" ns="/quadrupedal_test" />-->

        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"     
	respawn="false" output="screen">
	<remap from="/joint_states" to="/quadrupedal_test/joint_states" />
        </node>

        <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find quadrupedal_tests)/rviz/urdf.rviz" required="true" ns="/quadrupedal_test" /> -->
        
</launch>
